From: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Date: Thu, 11 Aug 2011 16:50:56 -0400
Subject: [PATCH 4/7] swiotlb: Expose swiotlb_nr_tlb function to modules

commit f21ffe9f6da6d3a69c518b7345c198d48d941c34 upstream.

As a mechanism to detect whether SWIOTLB is enabled or not.
We also fix the spelling - it was swioltb instead of
swiotlb.

CC: FUJITA Tomonori <fujita.tomonori@lab.ntt.co.jp>
[v1: Ripped out swiotlb_enabled]
Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
---
 drivers/xen/swiotlb-xen.c |    2 +-
 include/linux/swiotlb.h   |    2 +-
 lib/swiotlb.c             |    5 +++--
 3 files changed, 5 insertions(+), 4 deletions(-)

Index: linux-3.2.46/drivers/xen/swiotlb-xen.c
===================================================================
--- linux-3.2.46.orig/drivers/xen/swiotlb-xen.c	2013-07-26 18:53:35.000000000 +0000
+++ linux-3.2.46/drivers/xen/swiotlb-xen.c	2013-07-26 19:26:18.000000000 +0000
@@ -153,7 +153,7 @@
 	char *m = NULL;
 	unsigned int repeat = 3;
 
-	nr_tbl = swioltb_nr_tbl();
+	nr_tbl = swiotlb_nr_tbl();
 	if (nr_tbl)
 		xen_io_tlb_nslabs = nr_tbl;
 	else {
Index: linux-3.2.46/include/linux/swiotlb.h
===================================================================
--- linux-3.2.46.orig/include/linux/swiotlb.h	2013-07-26 18:53:35.000000000 +0000
+++ linux-3.2.46/include/linux/swiotlb.h	2013-07-26 19:26:18.000000000 +0000
@@ -24,7 +24,7 @@
 
 extern void swiotlb_init(int verbose);
 extern void swiotlb_init_with_tbl(char *tlb, unsigned long nslabs, int verbose);
-extern unsigned long swioltb_nr_tbl(void);
+extern unsigned long swiotlb_nr_tbl(void);
 
 /*
  * Enumeration for sync targets
Index: linux-3.2.46/lib/swiotlb.c
===================================================================
--- linux-3.2.46.orig/lib/swiotlb.c	2013-07-26 18:53:35.000000000 +0000
+++ linux-3.2.46/lib/swiotlb.c	2013-07-26 19:26:18.000000000 +0000
@@ -110,11 +110,11 @@
 __setup("swiotlb=", setup_io_tlb_npages);
 /* make io_tlb_overflow tunable too? */
 
-unsigned long swioltb_nr_tbl(void)
+unsigned long swiotlb_nr_tbl(void)
 {
 	return io_tlb_nslabs;
 }
-
+EXPORT_SYMBOL_GPL(swiotlb_nr_tbl);
 /* Note that this doesn't work with highmem page */
 static dma_addr_t swiotlb_virt_to_bus(struct device *hwdev,
 				      volatile void *address)
@@ -321,6 +321,7 @@
 		free_bootmem_late(__pa(io_tlb_start),
 				  PAGE_ALIGN(io_tlb_nslabs << IO_TLB_SHIFT));
 	}
+	io_tlb_nslabs = 0;
 }
 
 static int is_swiotlb_buffer(phys_addr_t paddr)
